### PR TITLE
CSI PowerFlex driver docs update - renameSDC and preapprovedGUID

### PR DESCRIPTION
**Pull request by khareRajshree**
# Description
Documentation update for CSI PowerFlex driver includes:

- Updates for rename SDC feature.
- Updates for Pre-approving SDC by GUID feature.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/402 |

# Checklist:

- [x] Have you run a grammar and spell checks against your submission?
- [ ] Have you tested the changes locally?
- [x] Have you tested whether the hyperlinks are working properly?
- [x] Did you add the examples wherever applicable?
- [ ] Have you added high-resolution images?

### CODE REVIEW

File: content/docs/csidriver/features/powerflex.md [hunk 1]
```diff
@@ -661,4 +661,63 @@ allowedTopologies:
         values:
           - csi-vxflexos.dellemc.com
 ```
-Once the volume gets created, the ControllerPublishVolume will set the QoS limits for the volumes mapped to SDC.
\ No newline at end of file
+Once the volume gets created, the ControllerPublishVolume will set the QoS limits for the volumes mapped to SDC.
+
+## Rename SDC
+
+Starting in version 2.6, CSI Driver for PowerFlex now supports renaming of SDC. To use this feature, the node section of values.yaml should have renameSDC keys enabled with a prefix value.
```
Comment (rsedlock1958) [inline content/docs/csidriver/features/powerflex.md:668]: Insert "the" before "CSI"

File: content/docs/csidriver/features/powerflex.md [hunk 2]
```diff
@@ -661,4 +661,63 @@ allowedTopologies:
...
+    #   false: disable renaming
+    # Default value: "false"
+    enabled: false
+    # "prefix" defines a string for the new name of the SDC.
+    # "prefix" + "worker_node_hostname" should not exceed 31 chars.
+    # Default value: none
+    # Examples: "rhel-sdc", "sdc-test"
+    prefix: "sdc-test"
+```
+The renameSDC section is going to be used by the Node Service, it has two keys enabled and prefix,
```
Comment (rsedlock1958) [inline content/docs/csidriver/features/powerflex.md:691]: Delete comma. Insert colon (":")

**Review by rsedlock1958 — COMMENTED** [review]
I have reviewed this. Several edits are required.

**Comment by khareRajshree** [comment]
> I have reviewed this. Several edits are required.

Hi @rsedlock1958, I have addressed the review comments.

File: content/docs/csidriver/features/powerflex.md (hunk 1 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: addressed

File: content/docs/csidriver/features/powerflex.md (hunk 2 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: addressed

File: content/docs/csidriver/features/powerflex.md [hunk 3]
```diff
@@ -661,4 +661,63 @@ allowedTopologies:
...
+* `enabled`: Boolean variable that specifies if the renaming for SDC is to be carried out or not. Is true then the driver will perform the rename operation. By default, its value will be false.
+* `prefix`: string variable that is used to set the prefix for SDC name.
+
+Now based on these two keys, there are certain scenarios on which the driver is going to perform the rename SDC operation:
+* If enabled and prefix given then set the prefix+worker_node_name for SDC name.
+* If enabled and prefix not given then set worker_node_name for SDC name.
+
+> NOTE: name of the SDC cannot be more than 31 characters, hence the prefix given and the worker node hostname name taken should be such that the total length do not exceed 31 character limit.
```
[the remark this answers is not in the document]
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: addressed

**Review by rsedlock1958 — APPROVED** [review]

**Comment by JacobGros** [comment]
please update the values.yaml section in install driver guide as well: https://github.com/dell/csm-docs/blob/main/content/docs/csidriver/installation/helm/powerflex.md#install-the-driver

File: content/docs/csidriver/features/powerflex.md [hunk 4]
```diff
@@ -661,4 +661,63 @@ allowedTopologies:
...
+
+Now based on these two keys, there are certain scenarios on which the driver is going to perform the rename SDC operation:
+* If enabled and prefix given then set the prefix+worker_node_name for SDC name.
+* If enabled and prefix not given then set worker_node_name for SDC name.
+
+> NOTE: name of the SDC cannot be more than 31 characters, hence the prefix given and the worker node hostname name taken should be such that the total length does not exceed 31 character limit. 
+
+## Pre-approving SDC by GUID
+
+Starting in version 2.6, the CSI Driver for PowerFlex now supports renaming the SDCs.
```
Comment (JacobGros) [inline content/docs/csidriver/features/powerflex.md:703]: This line should say approving sdc by guid, renaming SDC was previous section

**Review by JacobGros — CHANGES_REQUESTED** [review]
Please fix typo and update installation section with the new values

File: content/docs/csidriver/features/powerflex.md (hunk 4 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: Changes done.

**Comment by khareRajshree** [comment]
> Please fix typo and update installation section with the new values
Hi @JacobGros,  I have addressed the comments, please re-review it once.

**Review by JacobGros — APPROVED** [review]
LGTM

**Review by VamsiSiddu-7 — DISMISSED** [review]

File: content/docs/csidriver/features/powerflex.md [hunk 5]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
 ```
-Once the volume gets created, the ControllerPublishVolume will set the QoS limits for the volumes mapped to SDC.
\ No newline at end of file
+Once the volume gets created, the ControllerPublishVolume will set the QoS limits for the volumes mapped to SDC.
+
+## Rename SDC
+
+Starting in version 2.6, the CSI Driver for PowerFlex now supports renaming of SDC. To use this feature, the node section of values.yaml should have renameSDC keys enabled with a prefix value.
+
+If you want to rename the SDC, please make the following edits to your values file at `csi-vxflexos/helm/csi-vxflexos/values.yaml:`
```
Comment (rajkumar-palani) [inline content/docs/csidriver/features/powerflex.md:670]: provide link to the values.yaml file.

File: content/docs/csidriver/features/powerflex.md [hunk 6]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+* If enabled and prefix not given then set worker_node_name for SDC name.
+
+> NOTE: name of the SDC cannot be more than 31 characters, hence the prefix given and the worker node hostname name taken should be such that the total length does not exceed 31 character limit. 
+
+## Pre-approving SDC by GUID
+
+Starting in version 2.6, the CSI Driver for PowerFlex now supports pre-approving SDC by GUID.
+CSI PowerFlex driver should be able to detect/read the SDC mode from the PowerFlex array and determine whether requests for SDC approval be made to the array prior to publishing a volume. This is specific to each SDC.
+
+If you want to get the SDC pre-approved by GUID, please make the following edits to your values file at `csi-vxflexos/helm/csi-vxflexos/values.yaml:`
```
Comment (rajkumar-palani) [inline content/docs/csidriver/features/powerflex.md:707]: provide link to the values.yaml file.

File: content/docs/csidriver/features/powerflex.md [hunk 7]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+    # Default value: "false"
+    enabled: false
+    # "prefix" defines a string for the new name of the SDC.
+    # "prefix" + "worker_node_hostname" should not exceed 31 chars.
+    # Default value: none
+    # Examples: "rhel-sdc", "sdc-test"
+    prefix: "sdc-test"
+```
+The renameSDC section is going to be used by the Node Service, it has two keys enabled and prefix:
+* `enabled`: Boolean variable that specifies if the renaming for SDC is to be carried out or not. Is true then the driver will perform the rename operation. By default, its value will be false.
```
Comment (rajkumar-palani) [inline content/docs/csidriver/features/powerflex.md:693]: correct the typo from Is True to If True

File: content/docs/csidriver/features/powerflex.md [hunk 8]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+    # enabled: Enable/Disable SDC approval
+    #Allowed values:
+    #  true: Driver will attempt to approve restricted SDC by GUID during setup
+    #  false: Driver will not attempt to approve restricted SDC by GUID during setup
+    # Default value: false
+    enabled: false
+```
+> NOTE: Currently, the CSI-PowerFlex driver only supports GUID for the restricted SDC mode.
+
+If SDC approval is denied, then provisioning of the volume should not be attempted and the appropriate error message is reported in the logs/events so the user is informed.
```
Comment (rajkumar-palani) [inline content/docs/csidriver/features/powerflex.md:726]: change "the" to "an" appropriate.

**Review by rajkumar-palani — CHANGES_REQUESTED** [review]
please correct the review comments given.

File: content/docs/csidriver/features/powerflex.md (hunk 8 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md (hunk 7 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md (hunk 6 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md (hunk 5 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

**Comment by khareRajshree** [comment]
> please correct the review comments given.

Hi @rajkumar-palani,  have addressed the review comments.

File: content/docs/csidriver/features/powerflex.md [hunk 9]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
         values:
           - csi-vxflexos.dellemc.com
 ```
-Once the volume gets created, the ControllerPublishVolume will set the QoS limits for the volumes mapped to SDC.
\ No newline at end of file
+Once the volume gets created, the ControllerPublishVolume will set the QoS limits for the volumes mapped to SDC.
+
+## Rename SDC
+
+Starting in version 2.6, the CSI Driver for PowerFlex now supports renaming of SDC. To use this feature, the node section of values.yaml should have renameSDC keys enabled with a prefix value.
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:668]: Minor: CSI driver for PowerFlex will support renaming of SDCs.

File: content/docs/csidriver/features/powerflex.md [hunk 10]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+
+Now based on these two keys, there are certain scenarios on which the driver is going to perform the rename SDC operation:
+* If enabled and prefix given then set the prefix+worker_node_name for SDC name.
+* If enabled and prefix not given then set worker_node_name for SDC name.
+
+> NOTE: name of the SDC cannot be more than 31 characters, hence the prefix given and the worker node hostname name taken should be such that the total length does not exceed 31 character limit. 
+
+## Pre-approving SDC by GUID
+
+Starting in version 2.6, the CSI Driver for PowerFlex now supports pre-approving SDC by GUID.
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:704]: Minor: CSI Driver for PowerFlex will support pre-approving SDC by GUID.

File: content/docs/csidriver/features/powerflex.md [hunk 11]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+CSI PowerFlex driver should be able to detect/read the SDC mode from the PowerFlex array and determine whether requests for SDC approval be made to the array prior to publishing a volume. This is specific to each SDC.
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:705]: I would avoid "should", gives the impression that it is best try. I would rephrase as "CSI PowerFlex driver will detect the SDC mode set on the PowerFlex array and will request SDC approval from the array prior to publishing a volume"

File: content/docs/csidriver/features/powerflex.md [hunk 12]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+
+If you want to get the SDC pre-approved by GUID, please make the following edits to your [values.yaml](https://github.com/dell/csi-powerflex/blob/main/helm/csi-vxflexos/values.yaml) file:
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:707]: Change to : To request SDC approval for GUID, make the following edits...

File: content/docs/csidriver/features/powerflex.md [hunk 13]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+    # enabled: Enable/Disable SDC approval
+    #Allowed values:
+    #  true: Driver will attempt to approve restricted SDC by GUID during setup
+    #  false: Driver will not attempt to approve restricted SDC by GUID during setup
+    # Default value: false
+    enabled: false
+```
+> NOTE: Currently, the CSI-PowerFlex driver only supports GUID for the restricted SDC mode.
+
+If SDC approval is denied, then provisioning of the volume should not be attempted and an appropriate error message is reported in the logs/events so the user is informed.
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:726]: would avoid "should" again. If SDC approval is denied, then provisioning of the volume will not be attempted and an appropriate ....

**Review by bharathsreekanth — CHANGES_REQUESTED** [review]

File: content/docs/csidriver/features/powerflex.md (hunk 9 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md (hunk 10 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md (hunk 11 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md (hunk 12 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md (hunk 13 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

**Review by bharathsreekanth — CHANGES_REQUESTED** [review]

File: content/docs/csidriver/features/powerflex.md [hunk 14]
```diff
@@ -665,9 +665,9 @@ Once the volume gets created, the ControllerPublishVolume will set the QoS limit
 
 ## Rename SDC
 
-Starting in version 2.6, the CSI Driver for PowerFlex now supports renaming of SDC. To use this feature, the node section of values.yaml should have renameSDC keys enabled with a prefix value.
+Starting in version 2.6, the CSI driver for PowerFlex will support renaming of SDCs. To use this feature, the node section of values.yaml should have renameSDC keys enabled with a prefix value.
 
-If you want to rename the SDC, please make the following edits to your [values.yaml](https://github.com/dell/csi-powerflex/blob/main/helm/csi-vxflexos/values.yaml) file:
+To request renaming of SDC, make the following edits to [values.yaml](https://github.com/dell/csi-powerflex/blob/main/helm/csi-vxflexos/values.yaml) file:
```
[the remark this answers is not in the document]
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md [hunk 15]
```diff
@@ -665,9 +665,9 @@ Once the volume gets created, the ControllerPublishVolume will set the QoS limit
 
 ## Rename SDC
 
-Starting in version 2.6, the CSI Driver for PowerFlex now supports renaming of SDC. To use this feature, the node section of values.yaml should have renameSDC keys enabled with a prefix value.
+Starting in version 2.6, the CSI driver for PowerFlex will support renaming of SDCs. To use this feature, the node section of values.yaml should have renameSDC keys enabled with a prefix value.
```
[the remark this answers is not in the document]
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

File: content/docs/csidriver/features/powerflex.md [hunk 16]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+    # "prefix" + "worker_node_hostname" should not exceed 31 chars.
+    # Default value: none
+    # Examples: "rhel-sdc", "sdc-test"
+    prefix: "sdc-test"
+```
+The renameSDC section is going to be used by the Node Service, it has two keys enabled and prefix:
+* `enabled`: Boolean variable that specifies if the renaming for SDC is to be carried out or not. If true then the driver will perform the rename operation. By default, its value will be false.
+* `prefix`: string variable that is used to set the prefix for SDC name.
+
+Now based on these two keys, there are certain scenarios on which the driver is going to perform the rename SDC operation:
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:696]: You can remove "Now". Based on these two keys, the driver will perform the rename SDC operation as described below:

**Review by bharathsreekanth — APPROVED** [review]

File: content/docs/csidriver/features/powerflex.md (hunk 16 above)
Comment (khareRajshree) [reply content/docs/csidriver/features/powerflex.md]: done

Comment (bharathsreekanth) [reply content/docs/csidriver/features/powerflex.md]: Can you change it to "Based on these two keys, the driver will perform the rename SDC operation as described below", does it make sense?

**Review by bharathsreekanth — APPROVED** [review]

**Review by VamsiSiddu-7 — APPROVED** [review]

**Review by rajkumar-palani — APPROVED** [review]
lgtm

File: content/docs/csidriver/features/powerflex.md [hunk 17]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+    # enabled: Enable/Disable SDC approval
+    #Allowed values:
+    #  true: Driver will attempt to approve restricted SDC by GUID during setup
+    #  false: Driver will not attempt to approve restricted SDC by GUID during setup
+    # Default value: false
+    enabled: false
+```
+> NOTE: Currently, the CSI-PowerFlex driver only supports GUID for the restricted SDC mode.
+
+If SDC approval is denied, then provisioning of the volume will not be attempted and an appropriate error message is reported in the logs/events so the user is informed.
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:726]: minor comment, you can remove "so the user is informed" as purpose of logs/events is information to user. I will leave it up to you if you want to change it or leave as is.

File: content/docs/csidriver/features/powerflex.md [hunk 18]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
         values:
           - csi-vxflexos.dellemc.com
 ```
-Once the volume gets created, the ControllerPublishVolume will set the QoS limits for the volumes mapped to SDC.
\ No newline at end of file
+Once the volume gets created, the ControllerPublishVolume will set the QoS limits for the volumes mapped to SDC.
+
+## Rename SDC
+
+Starting with version 2.6, the CSI driver for PowerFlex will support renaming of SDCs. To use this feature, the node section of values.yaml should have renameSDC keys enabled with a prefix value.
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:668]: Do we need the last sentence? you are mentioning how to use the feature to rename SDC in the line below? upto you, if you want to change it.

File: content/docs/csidriver/features/powerflex.md [hunk 19]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+    #   false: disable renaming
+    # Default value: "false"
+    enabled: false
+    # "prefix" defines a string for the new name of the SDC.
+    # "prefix" + "worker_node_hostname" should not exceed 31 chars.
+    # Default value: none
+    # Examples: "rhel-sdc", "sdc-test"
+    prefix: "sdc-test"
+```
+The renameSDC section is going to be used by the Node Service, it has two keys enabled and prefix:
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:692]: Do we need to mention Node Service in the documentation? That is internal to our implementation right?
To enable renameSDC following keys are used.

File: content/docs/csidriver/features/powerflex.md [hunk 20]
```diff
@@ -661,4 +661,66 @@ allowedTopologies:
...
+* `enabled`: Boolean variable that specifies if the renaming for SDC is to be carried out or not. If true then the driver will perform the rename operation. By default, its value will be false.
```
Comment (bharathsreekanth) [inline content/docs/csidriver/features/powerflex.md:693]: I would rephrase as "Boolean variable that specified is renaming of SDC is desired". If true, then the driver will perform the rename operation. By default, the value is set to false.

**Review by bharathsreekanth — APPROVED** [review]
sorry Rajshree, I was looking at this again and made some comments. They are minor but I thought these will make the docs look crisper. I am approving the PR, but do consider the comments and see if it makes sense.
